stdlib/public/Concurrency/YieldingContinuation.swift

```diff
@@ -0,0 +1,196 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2020-2021 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+import Swift
+
+@available(macOS 9999, iOS 9999, watchOS 9999, tvOS 9999, *)
+public struct YieldingContinuation<Element, Failure: Error>: Sendable {
+  @_fixed_layout
+  @usableFromInline
+  internal final class Storage: UnsafeSendable {
+    @usableFromInline
+    var continuation: UnsafeContinuation<Element, Error>?
+  }
+  
+  @usableFromInline
+  let storage = Storage()
+
+  /// Construct a YieldingContinuation.
+  ///
+  /// This continuation type can be called more than once, unlike the unsafe and
+  /// checked counterparts. Each call to the yielding functions will resume any
+  /// awaiter on the next function. This type is inherently sendable and can
+  /// safely be used and stored in multiple task contexts.
+  public init() { }
+
+  /// Construct a YieldingContinuation with specific types including a failure.
+  ///
+  /// This continuation type can be called more than once, unlike the unsafe and
+  /// checked counterparts. Each call to the yielding functions will resume any
+  /// awaiter on the next function. This type is inherently sendable and can
+  /// safely be used and stored in multiple task contexts.
+  public init(yielding: Element.Type, throwing: Failure.Type) { }
+
+  @inlinable
+  @inline(__always)
+  internal func _extract() -> UnsafeContinuation<Element, Error>? {
+    let raw = Builtin.atomicrmw_xchg_seqcst_Word(
```

Comment:
This doesn't need to be `seqcst`.

Reply:
so which atomic op is correct for these? there doesn't seem to be an extensive documentation on which ones are available.

Reply:
changed to acqrel per the suggestion